tests/test_project.py

```diff
@@ -281,13 +297,30 @@ def test_find_jobs(self):
         assert 0 == len(list(self.project.find_jobs({"sp.a": 5})))
         assert 1 == len(list(self.project.find_jobs(doc_filter={"b": 0})))
         assert 0 == len(list(self.project.find_jobs(doc_filter={"b": 5})))
+        assert 1 == len(list(self.project.find_jobs(None, {"b": 0})))
+        assert 0 == len(list(self.project.find_jobs(None, {"b": 5})))
         assert 1 == len(list(self.project.find_jobs({"doc.b": 0})))
         assert 0 == len(list(self.project.find_jobs({"doc.b": 5})))
         assert 1 == len(list(self.project.find_jobs({"a": 0, "doc.b": 0})))
+        assert 1 == len(list(self.project.find_jobs({"a": 0}, {"b": 0})))
+        assert 1 == len(
+            list(self.project.find_jobs(filter={"a": 0}, doc_filter={"b": 0}))
+        )
+        assert 1 == len(list(self.project.find_jobs({"a": 0}, doc_filter={"b": 0})))
         assert 1 == len(list(self.project.find_jobs({"sp.a": 0, "doc.b": 0})))
+        assert 0 == len(list(self.project.find_jobs({"a": 0, "doc.b": 5})))
+        assert 0 == len(list(self.project.find_jobs({"a": 0}, {"b": 5})))
+        assert 0 == len(
+            list(self.project.find_jobs(filter={"a": 0}, doc_filter={"b": 5}))
+        )
+        assert 0 == len(list(self.project.find_jobs({"a": 0}, doc_filter={"b": 5})))
```

Comment:
In principle I think we should make sure to cover all combinations of:
`(None, ...)`, `({ .. }, ...)`, `(filter=None, ...)`, `(filter={ .. }, ...)`

and

`(..., None)`, `(..., { .. })`, `(..., doc_filter=None)`, `(..., doc_filter={ .. })`

Based on this, I believe you are missing: `(None, doc_filter={...})` and `(.., doc_filter=None)`.

Reply:
Do you mean that you want tests that explicitly pass `filter=None` and `doc_filter=None` instead of using the default parameter values of `None`?

Reply:
> Do you mean that you want tests that explicitly pass `filter=None` and `doc_filter=None` instead of using the default parameter values of `None`?

Yes, when it comes to this kind of manual arg and kwargs passing it is better to really test every possible combination. We might have gotten it right this time, but the next change might introduce a regression.

Reply:
Done in 92ccbffd4f3c65ad850fbe0c87d7be21d0437081.